app/models/policy_host.rb

```diff
@@ -10,10 +10,36 @@ class PolicyHost < ApplicationRecord
   validates :host_id, presence: true, uniqueness: { scope: :policy }
   validate :host_supported?, on: :create
 
-  def self.import_from_policy!(policy_id, host_ids)
-    import!(host_ids.map do |host_id|
-      { host_id: host_id, policy_id: policy_id }
-    end)
+  class << self
+    def import_from_policy!(policy_id, host_ids)
+      hosts = Host.where(id: host_ids).distinct
+      # Policy#find raises an exception if the record is not found
+      policy = Policy.find(policy_id)
+
+      # Bypass the validation if there are no hosts to be assigned
+      if hosts.any? && (!os_major_supported?(policy, hosts) || !os_minors_supported?(policy, hosts))
+        raise ActiveRecord::RecordInvalid.new(policy), 'Unsupported OS version in one of the hosts'
+      end
+
+      import(host_ids.map do |host_id|
+        { host_id: host_id, policy_id: policy_id }
+      end, validate: false, validate_uniqueness: true)
+    end
+
+    private
+
+    def os_major_supported?(policy, hosts)
+      os_major_version = hosts.pluck(Host::OS_MAJOR_VERSION)
```

Comment:
Although this is a private method, I'd still suggest to add `.distinct` here as well (just to be sure).